common/djangoapps/student/helpers.py

```diff
@@ -525,6 +527,9 @@ def _cert_info(user, course_overview, cert_status):
     ):
         return default_info
 
+    if not CourseMode.is_eligible_for_certificate(enrollment.mode, status=status):
+        return default_info
```

Comment:
Note: This was the primary change. We were not checking to see if enrollment was eligible for certs when looking at cert status for the Dashboard.